tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -75,36 +127,26 @@ void isr2(void)
  */
 void test_arm_irq_vector_table(void)
 {
-	printk("Test Cortex-M3 IRQ installed directly in vector table\n");
+	printk("Test Cortex-M IRQs installed directly in the vector table\n");
 
-	for (int ii = 0; ii < 3; ii++) {
-		irq_enable(ii);
-		_irq_priority_set(ii, 0, 0);
+	for (int ii = 0; ii < CONFIG_NUM_IRQS; ii++) {
+		irq_enable(_ISR_OFFSET + ii);
+		_irq_priority_set(_ISR_OFFSET + ii, 0, 0);
 		k_sem_init(&sem[ii], 0, UINT_MAX);
 	}
 
 	zassert_true((k_sem_take(&sem[0], K_NO_WAIT) ||
 		      k_sem_take(&sem[1], K_NO_WAIT) ||
 		      k_sem_take(&sem[2], K_NO_WAIT)), NULL);
 
-	for (int ii = 0; ii < 3; ii++) {
+	for (int ii = 0; ii < CONFIG_NUM_IRQS; ii++) {
 #if defined(CONFIG_SOC_TI_LM3S6965_QEMU)
 		/* the QEMU does not simulate the
 		 * STIR register: this is a workaround
 		 */
 		NVIC_SetPendingIRQ(ii);
```

Comment:
`_ISR_OFFSET + ` should be added here as well. Right now it is inconsistent with the loop enabling interrupts and setting their priorities. Although `_ISR_OFFSET` is `0` in this case, but...

Reply:
Added now. Yeah, it is a bit redundant for this platform, but, anyways...